lib/chef/provider/package/openbsd.rb

```diff
@@ -84,19 +90,28 @@ def installed_version
 
         def candidate_version
           @candidate_version ||= begin
-            version_string  = ''
-            version_string += "-#{version}" if @new_resource.version
-            pkg_info = shell_out!("pkg_info -I \"#{@new_resource.package_name}#{version_string}\"", :env => nil, :returns => [0,1])
-            if parts = @new_resource.package_name.match(/^(.+?)--(.+)/)
-              result = pkg_info.stdout[/^#{Regexp.escape(parts[1])}-(.+?)\s/, 1]
+            results = []
+            shell_out!("pkg_info -I \"#{@new_resource.package_name}#{version_string}\"", :env => nil, :returns => [0,1]).stdout.each_line do |line|
+              results << line[/^#{Regexp.escape(@new_resource.package_name)}-(.+?)\s/, 1]
+            end
+            results = results.reject(&:nil?)
+            Chef::Log.debug("candidate versions of '#{@new_resource.package_name}' are '#{results}'")
+            case results.length
+            when 0
+              []
+            when 1
+              results[0]
             else
-              result = pkg_info.stdout[/^#{Regexp.escape(@new_resource.package_name)}-(.+?)\s/, 1]
+              raise Chef::Exceptions::Package, "#{new_resource.name} has multiple matching candidates. You must specify a version" if results.length > 1
             end
-            Chef::Log.debug("candidate_version of '#{@new_resource.package_name}' is '#{result}'")
-            result
           end
         end
 
+        def version_string
+          ver  = ''
+          ver += "-#{new_resource.version}" if new_resource.version
```

Comment:
accessors are generally better for getters -- `new_resorce` will error out with nomethod, `@new_resorce` will autovivify an ivar.